extensions/SharkPool/BetterInput.js

```diff
@@ -0,0 +1,1305 @@
+                defaultValue: "input-url-or-uri-here",
+              },
+            },
+          },
+          {
+            blockType: Scratch.BlockType.LABEL,
+            text: "Effects",
+          },
+          {
+            opcode: "setEnterEffect",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "set enter effect to [ENTER_EFFECT]",
+            blockIconURI: effectIcon,
+            arguments: {
+              ENTER_EFFECT: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "enterEffectMenu",
+                defaultValue: "None",
+              },
+            },
+          },
+          {
+            opcode: "setEnterSpeed",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "set enter effect speed to [SPEED]",
+            blockIconURI: effectIcon,
+            arguments: {
+              SPEED: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "10",
+              },
+            },
+          },
+          {
+            opcode: "setExitEffect",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "set exit effect to [EXIT_EFFECT]",
+            blockIconURI: effectIcon,
+            arguments: {
+              EXIT_EFFECT: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "exitEffectMenu",
+                defaultValue: "None",
+              },
+            },
+          },
+          {
+            opcode: "setExitSpeed",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "set exit effect speed to [SPEED]",
+            blockIconURI: effectIcon,
+            arguments: {
+              SPEED: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: "10",
+              },
+            },
+          },
+          {
+            blockType: Scratch.BlockType.LABEL,
+            text: "Operations",
+          },
+          {
+            opcode: "setSubmitEvent",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "set force input to [ENTER]",
+            arguments: {
+              ENTER: {
+                type: Scratch.ArgumentType.STRING,
+                menu: "enterMenu",
+                defaultValue: "Disabled",
+              },
+            },
+          },
+          {
+            opcode: "setMaxBoxCount",
+            blockType: Scratch.BlockType.COMMAND,
+            text: "set max box count to: [MAX]",
+            arguments: {
+              MAX: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: 1,
+              },
+            },
+          },
+          {
+            opcode: "getBoxCount",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "box count",
+          },
+          {
+            opcode: "getMaxCount",
+            blockType: Scratch.BlockType.REPORTER,
+            text: "box limit",
+          },
+        ],
+        menus: {
```

Comment:
all of the menus should be acceptReporters: true unless they have a good reason not to